sonar-batch/src/main/java/org/sonar/batch/util/BatchUtils.java

```diff
@@ -48,4 +55,30 @@ public static String encodeForUrl(@Nullable String url) {
       throw new IllegalStateException("Encoding not supported", e);
     }
   }
+
+  public static String describe(Object o) {
+    try {
+      if (o.getClass().getMethod("toString").getDeclaringClass() != Object.class) {
+        return o.toString();
+      }
+    } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)